### PR TITLE
Use IBootstrap for the app

### PR DESCRIPTION
**Pull request by MorrisJobke**
Requires:

* [x] https://github.com/nextcloud/server/pull/21822
* [x] https://github.com/nextcloud/server/pull/21815

### CODE REVIEW

File: lib/Listener/LoadAdditionalScriptsListener.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+namespace OCA\FilesRightClick\Listener;
+
+use OCA\Files\Event\LoadAdditionalScriptsEvent;
+use OCA\Files_Sharing\Event\LoadAdditionalScriptsEvent as FilesSharingLoadAdditionalScriptsEvent;
+use OCP\EventDispatcher\Event;
+use OCP\EventDispatcher\IEventListener;
+
+class LoadAdditionalScriptsListener implements IEventListener {
+	public function handle(Event $event): void {
+		if (!($event instanceof LoadAdditionalScriptsEvent || $event instanceof FilesSharingLoadAdditionalScriptsEvent)) {
```
Comment (ChristophWurst) [inline lib/Listener/LoadAdditionalScriptsListener.php:36]: if you do not interact with the event (like use a getter of a prop or something) then this check isn't really necessary.

The reason I started adding it was to make the code *statically correct*. Like you always get the base `Event` and no specific sub class. Hence from a static perspective you can't assume a subtype. That changes if you check for it.

File: lib/Listener/LoadAdditionalScriptsLoggedInListener.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+namespace OCA\FilesRightClick\Listener;
+
+use OCP\AppFramework\Http\Events\LoadAdditionalScriptsLoggedInEvent;
+use OCP\EventDispatcher\Event;
+use OCP\EventDispatcher\IEventListener;
+
+class LoadAdditionalScriptsLoggedInListener implements IEventListener {
+	public function handle(Event $event): void {
+		if (!($event instanceof LoadAdditionalScriptsLoggedInEvent)) {
```
Comment (ChristophWurst) [inline lib/Listener/LoadAdditionalScriptsLoggedInListener.php:35]: same here

File: lib/Listener/LoadAdditionalScriptsListener.php (hunk 1 above)
Comment (MorrisJobke) [reply lib/Listener/LoadAdditionalScriptsListener.php:36]: I was also thinking about this. Yep we should then not do it.

**Comment by MorrisJobke** [comment]
Done 👍 Ready for review 🚀

**Pull request merged**